documentation/modules/configuring/proc-config-kafka-connect-s2i-migration.adoc

```diff
@@ -0,0 +1,85 @@
+// Module included in the following assemblies:
+//
+// assembly-kafka-connect.adoc
+
+[id='proc-migrating-kafka-connect-s2i-{context}']
+= Migrating from Kafka Connect with S2I to Kafka Connect
+
+[role="_abstract"]
+Support for Kafka Connect with S2I and the `KafkaConnectS2I` resource is planned for deprecation.
+The decision follows the introduction of `build` configuration properties to the `KafkaConnect` resource,
+which are used to build a container image with the connector plugins you require for your data connections automatically.
+
+This procedure describes how to migrate your Kafka Connect with S2I instance to a standard Kafka Connect instance.
+To do this, you configure a new `KafkaConnect` custom resource to replace the `KafkaConnectS2I` resource, which is deleted.
+
+WARNING: The migration process involves downtime from the moment `KafkaConnectS2I` instance is deleted until the new `KafkaConnect` instance has been successfully deployed.
+During this time, connectors will not be running and processing data. However, after the changeover they should continue from the point at which they stopped.
+
+.Prerequisites
+
+* Kafka Connect with S2I is deployed using a `KafkaConnectS2I` configuration
+* Kafka Connect with S2I is using an image with connectors added using an S2I build
+* Sink and source connector instances were created using `KafkaConnector` resources or the Kafka Connect REST API
+
+.Procedure
+
+. Create a new `KafkaConnect` custom resource using the same name as the name used for the `KafkaconnectS2I` resource.
+. Copy the `KafkaconnectS2I` resource properties to the `KafkaConnect` resource.
+. If specified, make sure you use the same `spec.config` properties:
++
+--
+* `group.id`
+* `offset.storage.topic`
+* `config.storage.topic`
+* `status.storage.topic`
++
+If these properties are not specified, defaults are used.
+In which case, leave them out of the `KafkaConnect` resource configuration as well.
+--
++
+Now add configuration specific to the `KafkaConnect` resource to the new resource.
+
+. Add `build` configuration to configure all the connectors and other libraries you want to add to the Kafka Connect deployment.
++
+NOTE: Alternatively, you can build a new image with connectors manually, and specify it using the `.spec.image` property.
+
+. Delete the old `KafkaConnectS2I` resource:
++
+[source,shell,subs="+quotes"]
+kubectl delete _MY-KAFKA-CONNECT-S2I-CONFIG-FILE_
```

Comment:
If we expect this to be a file, it should have `-f`:

 ```suggestion
kubectl delete -f _MY-KAFKA-CONNECT-S2I-CONFIG-FILE_
```

But the user might not have the file. So maybe we can add also `kubectl delete kafkaconnects2i _MY-KAFKA-CONNECT-S2I_`.

Reply:
Adding both sounds sensible.